psi4/driver/molutil.py

```diff
@@ -209,70 +59,80 @@ def molecule_get_attr(self, name):
+                         mass=None,
+                         real=None,
+                         elbl=None,
+
+                         name=None,
+                         units='Angstrom',
+                         input_units_to_au=None,
+                         fix_com=False,
+                         fix_orientation=False,
+                         fix_symmetry=None,
+
+                         fragment_separators=None,
+                         #fragment_types=None,
+                         fragment_charges=None,
+                         fragment_multiplicities=None,
+
+                         molecular_charge=None,
+                         molecular_multiplicity=None,
+
+                         zero_ghost_fragments=False,
+                         nonphysical=False,
+                         mtol=1.e-3,
+                         verbose=1,
+
+                         return_dict=False):
+        """Construct Molecule from unvalidated arrays and variables.
+
+        Light wrapper around :py:func:`~qcdb.molparse.from_arrays`
+        that is a full-featured constructor to dictionary representa-
+        tion of Molecule. This follows one step further to return
+        Molecule instance.
+
+        Parameters
+        ----------
+        See :py:func:`~qcdb.molparse.from_arrays`.
+
+        Returns
+        -------
+        :py:class:`psi4.core.Molecule`
+
+        """
+        molrec = qcdb.molparse.from_arrays(geom=geom,
+                                           elea=elea,
+                                           elez=elez,
+                                           elem=elem,
+                                           mass=mass,
+                                           real=real,
+                                           elbl=elbl,
+                                           name=name,
+                                           units=units,
+                                           input_units_to_au=input_units_to_au,
+                                           fix_com=fix_com,
+                                           fix_orientation=fix_orientation,
+                                           fix_symmetry=fix_symmetry,
+                                           fragment_separators=fragment_separators,
+                                           #fragment_types=fragment_types,
+                                           fragment_charges=fragment_charges,
+                                           fragment_multiplicities=fragment_multiplicities,
+                                           molecular_charge=molecular_charge,
+                                           molecular_multiplicity=molecular_multiplicity,
+                                           zero_ghost_fragments=zero_ghost_fragments,
+                                           nonphysical=nonphysical,
+                                           mtol=mtol,
+                                           verbose=verbose)
```

Comment:
Cool, can these take in list as well?

Reply:
Yes, all inputs are `array-like`. You can see it in action in [isapt1](https://github.com/psi4/psi4/pull/834/files#diff-c132b50739e2813b5bd3d74c5fb65d6aR52).

The output molrec dictionary is of fixed type, which is `ndarray` for atom stuff (geom, mass) and `list` for fragment stuff (fragment_separators, fragment_charges). I didn't quite understand your slack comment about which is preferable for the `to_json` operation (not in the scope of this PR).